base/hosts.c

```diff
@@ -1361,21 +1361,64 @@ gvm_hosts_resolve (gvm_hosts_t *hosts)
   gvm_hosts_deduplicate (hosts);
 }
 
+/**
+ * @brief Exclude a list of vhosts from a host's vhosts list.
+ *
+ * @param[in] host  The host whose vhosts are to be excluded from.
+ */
+void
+gvm_vhosts_exclude (gvm_host_t *host, const char *excluded_str)
+{
+  GSList *vhost;
+  char **excluded;
+
+  if (!host || !excluded_str)
+    return;
+
+  vhost = host->vhosts;
+  excluded = g_strsplit (excluded_str, ",", 0);
+  if (!excluded)
+    return;
+  while (vhost)
+    {
+      char **tmp = excluded;
+      char *value = ((gvm_vhost_t *) vhost->data)->value;
+
+      while (*tmp)
+        {
+          if (!strcmp (value, g_strstrip (*tmp)))
+            {
+              gvm_vhost_free (vhost->data);
+              host->vhosts = vhost = g_slist_delete_link (host->vhosts, vhost);
+              break;
+            }
+          tmp++;
+          if (!*tmp)
+            {
+              vhost = vhost->next;
+              break;
+            }
+        }
+    }
+  g_strfreev (excluded);
+
+  return;
+}
+
 /**
  * @brief Excludes a set of hosts provided as a string from a hosts collection.
  * Not to be used while iterating over the single hosts as it resets the
  * iterator.
  *
  * @param[in] hosts         The hosts collection from which to exclude.
  * @param[in] excluded_str  String of hosts to exclude.
- * @param[in] resolve       Boolean. Whether to also resolve hostnames when excluding.
  * @param[in] max_hosts     Max number of hosts in hosts_str. 0 means unlimited.
  *
  * @return Number of excluded hosts, -1 if error.
  */
 int
 gvm_hosts_exclude_with_max (gvm_hosts_t *hosts, const char *excluded_str,
-                            int resolve, unsigned int max_hosts)
```

Comment:
This change currently causing a build failure of GVM:

```
*redacted*/gvm/src/manage_utils.c:596:11: error: too many arguments to function ‘gvm_hosts_exclude_with_max’
       if (gvm_hosts_exclude_with_max (hosts,
           ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from *redacted*/gvm/src/manage_utils.c:33:
*redacted*/include/gvm/base/hosts.h:135:1: note: declared here
 gvm_hosts_exclude_with_max (gvm_hosts_t *, const char *, unsigned int);
 ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

https://github.com/greenbone/gvm/blob/v8.0%2Bbeta1/src/manage_utils.c#L596-L600

Reply:
@cfi-gb The related PR for gvm is https://github.com/greenbone/gvm/pull/198